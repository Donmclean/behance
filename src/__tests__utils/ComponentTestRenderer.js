import React from 'react';
import { store, history } from '../js/utils/createStore';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { mount, shallow } from 'enzyme';
import { Route } from 'react-router-dom';
import { ConnectedRouter } from 'react-router-redux';

const wrapComponent = (componentToRender) => (
    <Provider store={store}>
        <ConnectedRouter history={history}>
            <Route component={componentToRender} />
        </ConnectedRouter>
    </Provider>
);

export const getFullComponentJSON = (componentToRender) => {
    const componentWrapper = wrapComponent(componentToRender);
    return renderer.create((componentWrapper).toJSON());
};

export const getFullComponentRenderWrapper = (componentToRender) => {
    const componentWrapper = wrapComponent(componentToRender);
    return mount(componentWrapper);
};

export const getShallowComponentRendererWrapper = (componentToRender) => {
    const componentWrapper = wrapComponent(componentToRender);
    return shallow(componentWrapper);
};