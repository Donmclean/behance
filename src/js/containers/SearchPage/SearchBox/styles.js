import styled from 'styled-components';

export const SearchBoxWrapper = styled.div`
  height: 100%;
  width: 100%;
  text-align: center;
  margin: 150px 0;
`;

export const SearchBoxInput = styled.input`
    background: none;
    border: none;
    border-bottom: 2px solid black;
    text-align: center;
    width: 300px;
    font-size: 20px;
    outline: none;
`;