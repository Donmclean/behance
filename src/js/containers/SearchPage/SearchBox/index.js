import React from 'react';
import PropTypes from 'prop-types';
import { noop } from 'lodash';
import { SearchBoxWrapper, SearchBoxInput } from './styles'; 

const SearchBox = ({ searchValue = '', onChangeHandler = noop }) => (
    <SearchBoxWrapper>
        <SearchBoxInput value={searchValue} onChange={onChangeHandler} type="text" placeholder="Search for users.."/>
    </SearchBoxWrapper>
);

SearchBox.propTypes = {
    searchValue: PropTypes.string,
    onChangeHandler: PropTypes.func
};

export default SearchBox;
