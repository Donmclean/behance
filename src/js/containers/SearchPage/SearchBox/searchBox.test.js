import React from 'react';
import SearchBox from './index';

import { getFullComponentRenderWrapper } from '../../../../__tests__utils/ComponentTestRenderer';

describe('SearchBox Component', () => {
    it('renders <SearchBox /> component successfully in to the DOM with required props', () => {
        const props = {};

        const SearchBoxWithProps = () => (<SearchBox {...props} />);

        const tree = getFullComponentRenderWrapper(SearchBoxWithProps);

        expect(tree.debug()).toMatchSnapshot();
    });
});