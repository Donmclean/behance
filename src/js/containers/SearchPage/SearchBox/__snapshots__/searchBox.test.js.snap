// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox Component renders <SearchBox /> component successfully in to the DOM with required props 1`] = `
"<Provider store={{...}}>
  <ConnectedRouter history={{...}}>
    <Router history={{...}}>
      <Route component={[Function: SearchBoxWithProps]}>
        <SearchBoxWithProps match={{...}} location={{...}} history={{...}} staticContext={[undefined]}>
          <SearchBox>
            <styled.div>
              <div className=\\"sc-bdVaJa bxoFHy\\">
                <styled.input value=\\"\\" onChange={[Function: noop]} type=\\"text\\" placeholder=\\"Search for users..\\">
                  <input className=\\"sc-bwzfXH iWxmTX\\" value=\\"\\" onChange={[Function: noop]} type=\\"text\\" placeholder=\\"Search for users..\\" />
                </styled.input>
              </div>
            </styled.div>
          </SearchBox>
        </SearchBoxWithProps>
      </Route>
    </Router>
  </ConnectedRouter>
</Provider>"
`;
